express_relay/sdk/js/src/examples/simpleSearcherLimo.ts

```diff
@@ -112,41 +119,49 @@ class SimpleSearcherLimo {
       order.address,
       bidAmount,
       new anchor.BN(Math.round(Date.now() / 1000 + DAY_IN_SECONDS)),
-      this.chainId
+      this.chainId,
+      this.expressRelayConfig!.relayerSigner,
+      this.expressRelayConfig!.feeReceiverRelayer
     );
 
+    bid.transaction.recentBlockhash = opportunity.blockHash;
+    bid.transaction.sign(this.searcher);
+    return bid;
+  }
+
+  async opportunityHandler(opportunity: Opportunity) {
+    if (!("order" in opportunity))
+      throw new Error("Not a valid SVM opportunity");
+    const bid = await this.generateBid(opportunity);
     try {
-      const { blockhash } = await this.connectionSvm.getLatestBlockhash();
-      bid.transaction.recentBlockhash = blockhash;
-      bid.transaction.sign(this.searcher);
       const bidId = await this.client.submitBid(bid);
-      console.log(`Successful bid. Bid id ${bidId}`);
+      console.log(
+        `Successful bid. Opportunity id ${opportunity.opportunityId} Bid id ${bidId}`
+      );
     } catch (error) {
-      console.error(`Failed to bid: ${error}`);
+      console.error(
+        `Failed to bid on opportunity ${opportunity.opportunityId}: ${error}`
+      );
     }
   }
 
-  async bidOnNewOrders() {
-    let allOrders =
-      await this.clientLimo.getAllOrdersStateAndAddressWithFilters([]);
-    allOrders = allOrders.filter(
-      (order) => !order.state.remainingInputAmount.isZero()
+  async fetchConfig() {
+    this.expressRelayConfig = await this.client.getExpressRelaySvmConfig(
+      this.chainId,
+      this.connectionSvm
     );
-    if (allOrders.length === 0) {
-      console.log("No orders to bid on");
-      return;
-    }
-    for (const order of allOrders) {
-      await this.evaluateOrder(order);
-    }
-    // Note: You need to parallelize this in production with something like:
-    // await Promise.all(allOrders.map((order) => this.evaluateOrder(order)));
   }
 
   async start() {
-    for (;;) {
-      await this.bidOnNewOrders();
-      await new Promise((resolve) => setTimeout(resolve, 2000));
+    await this.fetchConfig();
+    try {
+      await this.client.subscribeChains([argv.chainId]);
```

Comment:
I think if we pass a `chain_type` to server with `chain_id`, we can remove bunch of type checkings above :-?

Reply:
true, assuming we want to keep stuff backward compatible and not change a lot on the server side, I thought this is a good enough solution